### PR TITLE
OPA: Add policy conception

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: denis-tingajkin <denis.tingajkin@xored.com>

## Motivation

1. Added abstraction layer for rego query executor
```go 
package opa

type AuthorizationPolicy interface {
	Check(ctx context.Context, input interface{}) error
}
func WithPolicyFromSource(source, query string, checkQuery CheckQueryFunc)AuthorizationPolicy
func WithPolicyFromFile(path, query string, checkQuery CheckQueryFunc)AuthorizationPolicy
```
2.  Added a possibility to use a set of authorization policies for pkg `authorize`.
```go
package authorize

func NewServer(policies ...opa.AuthorizationPolicy) networkservice.NetworkServiceServer
```
3. Added tests to cover.

### CODE REVIEW

**Comment by denis-tingajkin** [comment]
Found a few issues, will fix

**Comment by edwarnicke** [comment]
Oh HELL Yes!

Looks like its based on other PRs that need to go in first though :)

File: pkg/networkservice/chains/endpoint/server.go [hunk 1]
```diff
@@ -53,14 +51,15 @@ type endpoint struct {
 // NewServer - returns a NetworkServiceMesh client as a chain of the standard Client pieces plus whatever
 //             additional functionality is specified
 //             - name - name of the NetworkServiceServer
+//             - authServer authorization server chain element
 //             - tokenGenerator - token.GeneratorFunc - generates tokens for use in Path
 //             - additionalFunctionality - any additional NetworkServiceServer chain elements to be included in the chain
-func NewServer(name string, authzPolicy *rego.PreparedEvalQuery, tokenGenerator token.GeneratorFunc, additionalFunctionality ...networkservice.NetworkServiceServer) Endpoint {
+func NewServer(name string, authServer networkservice.NetworkServiceServer, tokenGenerator token.GeneratorFunc, additionalFunctionality ...networkservice.NetworkServiceServer) Endpoint {
```
Comment (edwarnicke) [inline pkg/networkservice/chains/endpoint/server.go:57]: please make this authz ... so as not to confuse it with authn :)

Comment (denis-tingajkin) [reply pkg/networkservice/chains/endpoint/server.go]: Comment above says that it is authorization server :)
But ok, fixed.

File: pkg/networkservice/chains/endpoint/server.go [hunk 2]
```diff
@@ -53,14 +51,15 @@ type endpoint struct {
 // NewServer - returns a NetworkServiceMesh client as a chain of the standard Client pieces plus whatever
 //             additional functionality is specified
 //             - name - name of the NetworkServiceServer
+//             - authServer authorization server chain element
```
Comment (edwarnicke) [inline pkg/networkservice/chains/endpoint/server.go]: authzServer

File: pkg/networkservice/chains/nsmgr/server.go [hunk 3]
```diff
@@ -50,13 +50,13 @@ type nsmgr struct {
 
 // NewServer - Creates a new Nsmgr
 //           name - name of the Nsmgr
-//           authzPolicy - authorization policy
+//           authServer - authorization server chain element
 //           registryCC - client connection to reach the upstream registry
-func NewServer(name string, authzPolicy *rego.PreparedEvalQuery, tokenGenerator token.GeneratorFunc, registryCC grpc.ClientConnInterface) Nsmgr {
+func NewServer(name string, authServer networkservice.NetworkServiceServer, tokenGenerator token.GeneratorFunc, registryCC grpc.ClientConnInterface) Nsmgr {
```
Comment (edwarnicke) [inline pkg/networkservice/chains/nsmgr/server.go]: authzServer

File: pkg/networkservice/chains/nsmgr/server.go [hunk 4]
```diff
@@ -50,13 +50,13 @@ type nsmgr struct {
...
 	rv := &nsmgr{}
 	rv.Endpoint = endpoint.NewServer(
 		name,
-		authzPolicy,
+		authServer,
```
Comment (edwarnicke) [inline pkg/networkservice/chains/nsmgr/server.go]: authzServer

File: pkg/networkservice/chains/endpoint/server.go (hunk 2 above)
Comment (denis-tingajkin) [reply pkg/networkservice/chains/endpoint/server.go]: already fixed.

File: pkg/networkservice/chains/nsmgr/server.go (hunk 3 above)
Comment (denis-tingajkin) [reply pkg/networkservice/chains/nsmgr/server.go]: already fixed.

File: pkg/networkservice/chains/nsmgr/server.go (hunk 4 above)
Comment (denis-tingajkin) [reply pkg/networkservice/chains/nsmgr/server.go]: already fixed.

**Pull request merged**